src/D_functions_and_procedures.sql

```diff
@@ -473,32 +437,63 @@ END;
 $$ LANGUAGE plpgsql;
 
 -- D3 TEST
--- Insert Sample Data
-INSERT INTO media (type, plot, runtime, imdb_id) 
-VALUES 
-('Movie', 'A mind-bending thriller about dreams and reality.', 148, 'tt1375666');
-
-INSERT INTO "user" (username, password, email) VALUES 
-('john_doe', 'password123', 'john@example.com'),
-('jane_doe', 'password456', 'jane@example.com')
-ON CONFLICT (username) DO NOTHING;
-
--- Test Rating Functionality
 DO $$
 DECLARE
     john_id INTEGER;
     jane_id INTEGER;
 BEGIN
     -- Retrieve user IDs for john_doe and jane_doe
     SELECT user_id INTO john_id FROM "user" WHERE username = 'john_doe';
-    SELECT user_id INTO jane_id FROM "user" WHERE username = 'jane_doe';
+    SELECT user_id INTO jane_id FROM "user" WHERE username = 'jane_smith';
 
     -- Perform ratings using the rate() function
-    PERFORM rate(john_id, 'tt1375666', 8.0); 
-    PERFORM rate(john_id, 'tt1375666', 9.0); 
-    PERFORM rate(jane_id, 'tt1375666', 7.0); 
+    PERFORM rate(john_id, 'tt1375666', 8.0, 'nice'); 
+    PERFORM rate(john_id, 'tt1375666', 9.0, 'super nice'); 
+    PERFORM rate(jane_id, 'tt1375666', 7.0, 'okay');
+	PERFORM rate(jane_id, 'tt13729548', 2.0, 'bad');
 END $$;
```

Comment:
Should all these things be moved to the test script for the hand in?

Reply:
Yes I would believe so, as far as I understand it the functions script file should only contain the creation of functions.